airgun/entities/hostgroup.py

```diff
@@ -74,6 +74,45 @@ def total_no_of_assigned_role(self, entity_name):
         wait_for(lambda: int(role_list[-1].text.split(". ")[0]), timeout=30)
         return int(role_list[-1].text.split(". ")[0])
 
+    def assign_role_to_hostgroup(self, entity_name, role_name):
+        """Assign Ansible role(s) to the host group based on user input
+        Args:
+            entity_name: Name of the host
+            role_name: Name of the ansible role
+        """
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.ansible_roles.more_item.click()
+        view.ansible_roles.select_pages.click()
+        role_list = self.browser.elements(view.ansible_roles.available_role, parent=self)
+        for role in role_list[1:]:
+            if role.text.split(". ")[1] == role_name:
+                role.click()
+        view.submit.click()
+
+    def remove_hostgroup_role(self, entity_name, role_name):
+        """Remove Ansible role from the host group based on user input
+        Args:
+            entity_name: Name of the host
+            role_name: Name of the ansible role
+        """
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.ansible_roles.click()
+        role_list = self.browser.elements(view.ansible_roles.assigned_role, parent=self)
+        for role in role_list:
+            if role.text.split(". ")[1] == role_name:
+                role.click()
+        view.submit.click()
+
+    def read_role(self, entity_name, role_name):
+        """Return name of the assigned Ansible role(s) of the host group."""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.ansible_roles.click()
+        role_list = self.browser.elements(view.ansible_roles.assigned_role, parent=self)
+        role = [
+            role.text.split(". ")[1] for role in role_list if role.text.split(". ")[1] in role_name
+        ]
+        return role
```

Comment:
move these into the `MultiSelectNoFilter` widget for something as read_assigned_values(), this can be used for other vide as well.